piksi_tools/fileio.py

```diff
@@ -470,21 +479,24 @@ def main():
             f = FileIO(link)
             try:
                 if args.write:
-                    f.write(args.write[1], open(args.write[0]).read())
+                    f.write(raw_filename(args.write[1]), bytearray(open(args.write[0], 'rb').read()))
                 elif args.read:
                     if len(args.read) not in [1, 2]:
                         sys.stderr.write("Error: fileio read requires either 1 or 2 arguments, SOURCE and optionally DEST.")
                         sys.exit(1)
-                    data = f.read(args.read[0])
+                    data = f.read(raw_filename(args.read[0]))
                     if len(args.read) == 2:
-                        with open(args.read[1], ('wb' if args.hex else 'w')) as fd:
+                        with open(args.read[1], ('w' if args.hex else 'wb')) as fd:
```

Comment:
Why the switch?

Reply:
Because I think the original order was actually incorrect: if we want to read a binary file from the device and write a copy locally as a hexdump, the output is a text file but if we want to write the local copy in original binary format, it's binary. This didn't actually matter in python2 because a text string could contain binary data that could be written into a file open in text mode but in python3 it matters.

Reply:
👍